From 69f05c50c5b6ef960a8dcc5790bba4a916180a71 Mon Sep 17 00:00:00 2001
From: Iain Lane <iain@orangesquash.org.uk>
Date: Wed, 20 Sep 2017 15:01:41 +0100
Subject: [PATCH 1/3] manager: Disconnect from signals on the proxy when we're
 disposed

We're calling a callback on a proxy after it has been disposed. We
should make sure to disconnect from it when we go away.

https://bugzilla.gnome.org/show_bug.cgi?id=787897
(cherry picked from commit 91fa202379b417f146c9b14be7b9b08d2dc83a78)
(cherry picked from commit 297862b0e9a6b08e506a9976f2e464f68ba398d2)
---
 libnm/nm-manager.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libnm/nm-manager.c b/libnm/nm-manager.c
index d3df73375..294b9de18 100644
--- a/libnm/nm-manager.c
+++ b/libnm/nm-manager.c
@@ -192,8 +192,8 @@ init_dbus (NMObject *object)
 	                                property_info);
 
 	/* Permissions */
-	g_signal_connect (priv->proxy, "check-permissions",
-	                  G_CALLBACK (manager_recheck_permissions), object);
+	g_signal_connect_object (priv->proxy, "check-permissions",
+				 G_CALLBACK (manager_recheck_permissions), object, 0);
 }
 
 static NMClientPermission
-- 
2.13.6

